backend/functions/lib/adapters/gateways/case.assignment.local.repository.ts

```diff
@@ -0,0 +1,49 @@
+import { ICaseAssignmentRepository } from '../../interfaces/ICaseAssignmentRepository';
+import { CaseAttorneyAssignment } from '../types/case.attorney.assignment';
+import { ApplicationContext } from '../types/basic';
+
+export class CaseAssignmentLocalRepository implements ICaseAssignmentRepository {
+  private caseAttorneyAssignments: CaseAttorneyAssignment[] = [];
+  private _nextUnusedId = 1;
+
+  public async createAssignment(
+    context: ApplicationContext,
+    caseAssignment: CaseAttorneyAssignment,
+  ): Promise<number> {
+    const assignment: CaseAttorneyAssignment = await this.findAssignment(caseAssignment);
```

Comment:
If an assignment has already been created on a case, and another assignment request for the same case comes in, respond with a 400 bad request.